pkg/cluster/workloadidentityresources_test.go

```diff
@@ -764,13 +775,234 @@ func TestGetPlatformWorkloadIdentityFederatedCredName(t *testing.T) {
+				},
+			},
+		},
+		{
+			name: "new required identity added - creates new roleassignment",
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					ClusterProfile: api.ClusterProfile{
+						ResourceGroupID: resourceGroupID,
+					},
+					PlatformWorkloadIdentityProfile: &api.PlatformWorkloadIdentityProfile{
+						PlatformWorkloadIdentities: map[string]api.PlatformWorkloadIdentity{
+							"1": {
+								ObjectID: objectId1,
+							},
+							"2": {
+								ObjectID: objectId2,
+							},
+						},
+					},
+				},
+			},
+			roles: map[string]api.PlatformWorkloadIdentityRole{
+				"1": {
+					OperatorName:     "1",
+					RoleDefinitionID: roleDefinitionId1,
+				},
+				"2": {
+					OperatorName:     "2",
+					RoleDefinitionID: roleDefinitionId2,
+				},
+			},
+			existingRoleAssignments: []mgmtauthorization.RoleAssignment{
+				{
+					Name: &roleName1,
+					RoleAssignmentPropertiesWithScope: &mgmtauthorization.RoleAssignmentPropertiesWithScope{
+						Scope:            &resourceGroupID,
+						RoleDefinitionID: pointerutils.ToPtr(fmt.Sprintf("/subscriptions/%s%s", subscriptionId, roleDefinitionId1)),
+						PrincipalID:      &objectId1,
+					},
+				},
+			},
+			wantAdded: []*arm.Resource{
+				workloadIdentityResourceGroupRBAC(stringutils.LastTokenByte(roleDefinitionId2, '/'), objectId2),
+			},
+		},
+		{
+			name: "identity object id replaced - deletes old roleassignment and creates new",
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					ClusterProfile: api.ClusterProfile{
+						ResourceGroupID: resourceGroupID,
+					},
+					PlatformWorkloadIdentityProfile: &api.PlatformWorkloadIdentityProfile{
+						PlatformWorkloadIdentities: map[string]api.PlatformWorkloadIdentity{
+							"1": {
+								ObjectID: objectId2,
+							},
+						},
+					},
+				},
+			},
+			roles: map[string]api.PlatformWorkloadIdentityRole{
+				"1": {
+					OperatorName:     "1",
+					RoleDefinitionID: roleDefinitionId1,
+				},
+			},
+			existingRoleAssignments: []mgmtauthorization.RoleAssignment{
+				{
+					Name: &roleName1,
+					RoleAssignmentPropertiesWithScope: &mgmtauthorization.RoleAssignmentPropertiesWithScope{
+						Scope:            &resourceGroupID,
+						RoleDefinitionID: pointerutils.ToPtr(fmt.Sprintf("/subscriptions/%s%s", subscriptionId, roleDefinitionId1)),
+						PrincipalID:      &objectId1,
+					},
+				},
+			},
+			wantDeleted: []mgmtauthorization.RoleAssignment{
+				{
+					Name: &roleName1,
+					RoleAssignmentPropertiesWithScope: &mgmtauthorization.RoleAssignmentPropertiesWithScope{
+						Scope:            &resourceGroupID,
+						RoleDefinitionID: pointerutils.ToPtr(fmt.Sprintf("/subscriptions/%s%s", subscriptionId, roleDefinitionId1)),
+						PrincipalID:      &objectId1,
+					},
+				},
+			},
+			wantAdded: []*arm.Resource{
+				workloadIdentityResourceGroupRBAC(stringutils.LastTokenByte(roleDefinitionId1, '/'), objectId2),
+			},
+		},
+	} {
```

Comment:
We are not concerned about a deprecated roleDefinition, correct?

Reply:
What would that look like? I haven't really thought about that scenario, but if we e.g. replace a role definition for a specific workload with a different one, this code should still handle this as the "wrong role assignment on the identity" and replace it, right?

Reply:
I am not sure too how it is being handled exactly through out the flow, but when just focusing on the role assignment deletion, the iteration takes place on the roles present, and if the deprecated role is not present in the list of roles for the version that roleAssignment won't go in the toDelete part. So the roleAssignments for deprecated roles would remain hanging.